server/blob/blob-deduplicating/src/test/scala/GCPropertiesTest.scala

```diff
@@ -0,0 +1,216 @@
+/***************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.server.blob.deduplication
+
+import java.nio.charset.StandardCharsets
+
+import com.google.common.hash
+import org.apache.james.blob.api.BlobId
+import org.apache.james.server.blob.deduplication.Generators.{OnePassGCTestParameters, TestParameters}
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Test.Parameters
+import org.scalacheck.{Arbitrary, Gen, Properties, Shrink}
+
+case class GenerationAwareBlobId(generation: Generation, hash: String) extends BlobId {
```

Comment:
Great. But this very important concept should at least have a trait and be part of production code.